daemon/oci_linux.go

```diff
@@ -171,64 +170,45 @@ func WithCapabilities(c *container.Container) coci.SpecOpts {
 	}
 }
 
-func readUserFile(c *container.Container, p string) (io.ReadCloser, error) {
-	fp, err := c.GetResourcePath(p)
+func resourcePath(c *container.Container, getPath func() (string, error)) (string, error) {
+	p, err := getPath()
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	fh, err := os.Open(fp)
-	if err != nil {
-		// This is needed because a nil *os.File is different to a nil
-		// io.ReadCloser and this causes GetExecUser to not detect that the
-		// container file is missing.
-		return nil, err
-	}
-	return fh, nil
+	return c.GetResourcePath(p)
 }
 
-func getUser(c *container.Container, username string) (uint32, uint32, []uint32, error) {
-	passwdPath, err := user.GetPasswdPath()
+func getUser(c *container.Container, username string) (specs.User, error) {
```

Comment:
If you would use named return variables, you won't have to repeat `return specs.User{}` a few times.

Reply:
optional of course; I just think it will look a tad cleaner that way

Reply:
Yes, I was also looking a bit at that; considering a `var .. specs.User` at the start of the function (not a huge fan of named return) or to make it a pointer so that `nil` can be returned.

Let me try a bit to see what looks best